server/src/main/resources/application.properties

```diff
@@ -1 +1,13 @@
 spring.data.mongodb.database=cards
+
+spring.security.oauth2.client.registration.spring.provider=spring
+spring.security.oauth2.client.registration.spring.client-id=flashcards
+spring.security.oauth2.client.registration.spring.client-secret=secret
+spring.security.oauth2.client.registration.spring.client-authentication-method=client_secret_basic
+spring.security.oauth2.client.registration.spring.authorization-grant-type=authorization_code
+spring.security.oauth2.client.registration.spring.redirect-uri={baseUrl}/login/oauth2/code/{registrationId}
```

Comment:
I'm not very proficient with oauth, could you please explain where do these {baseUrl} and {registrationId} come from or go to?

Reply:
https://docs.spring.io/spring-security/reference/servlet/oauth2/login/core.html#oauth2login-sample-redirect-uri
see the tipp there. baseUrl is not really explained there though. registrationId is the registration token of oauth2 (negotiated in authentication process)
I guess the baseUrl is filled by the ServletContext (Tomcat)